pkg/proxy/proxy.go

```diff
@@ -295,3 +303,15 @@ func (p *Proxy) GetStatusModel() *models.ProxyStatus {
 		PortRange: fmt.Sprintf("%d-%d", p.rangeMin, p.rangeMax),
 	}
 }
+
+// UpdateRedirectMetrics updates the redirect metrics per application protocol
+// in Prometheus. Lock needs to be held to call this function.
+func (p *Proxy) UpdateRedirectMetrics() {
+	result := map[string]int{}
+	for _, redirect := range p.redirects {
+		result[string(redirect.parserType)]++
+	}
+	for proto, count := range result {
+		metrics.ProxyRedirects.WithLabelValues(proto).Set(float64(count))
+	}
```

Comment:
I think we can have just one forloop like:

```
for _, redirect := range p.redirects {
    metrics.ProxyRedirects.WithLabelValues(string(redirect.parserType)).Inc()
}
```

Then we won't need to store those values in additional map.

Reply:
This will require changing the type of metric from gauge to counter.

Reply:
Hi, 

The reason why a gauge is used to keep track of how many redirects without using the `rate` function in Prometheus, this is a number that it's global per cilium and the number of redirects is always fixed, and does not increment per events.